File: scripts/compiled/colors.c
#include <stdio.h>

// ANSI escape codes for 256 colors
#define FG_COLOR "\x1b[38;5;%dm"
#define BG_COLOR "\x1b[48;5;%dm"
#define RESET_COLOR "\x1b[0m"

int main() {
    for (short i = 0; i <= 15; i++) {
        printf(FG_COLOR "Color %2d" RESET_COLOR "   " BG_COLOR "  Color %2d  " RESET_COLOR "\n", i, i, i, i);
    }

    return 0;
}

File: flake/nixOsModules/customScripts/colors.rs
use std::{
    io::{Write, stdout},
    env::args
};

const FG_COLOR: &str = "\x1b[38;5;";
const BG_COLOR: &str = "\x1b[48;5;";
const RESET_COLOR: &str = "\x1b[0m";

fn main() {
    let upper_limit: u8 = args()
        .nth(1)
        .and_then(|n| n.parse().ok())
        .unwrap_or(15);
    for i in 0..=upper_limit {
        println!(
            "{FG_COLOR}{i}mColor {i:2}{RESET_COLOR}   {BG_COLOR}{i}m  Color {i:2}  {RESET_COLOR}",
        );
    }

    _ = stdout().flush();
}
